doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,228 @@
+
+.. list-table::
+   :widths: 35 45 60 
+   :header-rows: 1
+
+   * - **Device**
+     - **Shortname**
+     - **Description**
+   * - `IBM Q Experience <https://pennylaneqiskit.readthedocs.io/en/latest/devices/ibmq.html>`__
+     - ``"qiskit.ibmq"`` *(must specify a hardware backend)*
+     - IBM Q hardware device, queue based access to IBMQ backends
+   * - `Forest QPUDevice <https://pennylane-forest.readthedocs.io/en/latest/code/qpu.html>`__
+     - ``"forest.qpu"``
+     - Forest QPU device, session based access to Rigetti QPUs
+
+:html:`</div></div>`
+
+:html:`<div class="photonic_list" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content4" class="collapsed"><p class="first admonition-title">Photonic devices (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content4" class="collapse" data-parent="#aside1" style="height: 0px;">`
+
+.. list-table::
+   :widths: 35 45 60 
+   :header-rows: 1
+
+   * - **Device**
+     - **Shortname**
+     - **Description**
+   * - :class:`~.DefaultGaussian`
+     - ``"default.gaussian"``
+     - Default gaussian device
+   * - `Strawberry Fields Fock device <https://pennylane-sf.readthedocs.io/en/latest/devices/fock.html>`__
+     - ``"strawberryfields.fock"``
+     - Fock device giving access to the Strawberry Fields Fock state simulator backend
+   * - `Strawberry Fields Gaussian device <https://pennylane-sf.readthedocs.io/en/latest/devices/gaussian.html>`__
+     - ``"strawberryfields.gaussian"``
+     - Gaussian device giving access to the Strawberry Fields Fock state simulator backend
+
+
+:html:`</div></div>`
+
+QNodes
+******
+
+A  quantum  node or ``QNode`` (represented by a subclass to
+:class:`~.BaseQNode`) is an encapsulation of a function :math:`f(x;θ):R^m→R^n`
+that is executed using quantum information processing on a quantum
+device.
+
+Each ``QNode`` represents the quantum circuit by building a
+:class:`~.CircuitGraph` instance, but the way differentiation is done is custom
+to the differentiation method offered by the ``QNode``.
+
+For further details on qnodes, the mutability of qnodes we refer to
+:class:`~.BaseQNode` and for a list of qnodes to :ref:`qml_qnodes`.
+
+Interfaces
+**********
+
+The integration between classical and quantum computations is encompassed by
+interfaces.
+
+We refer to the :ref:`intro_interfaces` page for a more in-depth introduction
+and a list of available interfaces.
+
+Optimizers
+**********
+
+Optimizers are objects which can be used to automatically update the parameters
+of a quantum or hybrid machine learning model.
+
+We refer to the :ref:`intro_ref_opt` page for a more in-depth introduction
+and a list of available optimizers.
+
+Key design details
+##################
+
+The following are key design details related to how PennyLane works internally.
+
+Queuing of operators
+********************
```

Comment:
I think this is great to include! However, we could potentially include more specifics; e.g.,

* Operations are queued on initialization, via a call to `Operation.queue()`

* Already instantiated operations cannot be re-queued unless `op.queue()` is manually called within the queueing context.

* `Observables` do not queue themselves; the _measurement functions_ `qml.expval`, `qml.var` etc. do the queueing. Hence, observables _can_ be instantiated before queueing.

It also seems a good place to mention the `OperationRecorder`.

Reply:
Great points, added them! :100: